app/models/game.rb

```diff
@@ -48,6 +48,47 @@ def populate_board! # rubocop:disable Metrics/AbcSize
     pieces.create(color: 'black', type: 'King', current_row_index: 7, current_column_index: 4)
   end
 
+  def checkmate?(color)
+    return false unless in_check?(color)
+    # Determine if king can move out of check to escape
+    checked_king = pieces.find_by_type_and_color(King, color)
+    return false if checked_king.can_move_out_of_check?
+    # here threatening piece is opposite color of checked king
+    return false if @threatening_piece.can_be_captured?
+    return false if @threatening_piece.can_be_blocked?(checked_king)
+    true
+  end
+
+  def in_check?(color)
+    king = pieces.find_by_type_and_color(King, color)
+    opponent_pieces = opposite_remaining_pieces_of(color)
+    opponent_pieces.each do |piece|
+      if piece.valid_move?(king.current_row_index, king.current_column_index)
+        @threatening_piece = piece
+        return true
+      end
+    end
+    false
+  end
```

Comment:
This looks great.

One improvement I have for you, is trying to remove the instance variable `@threatening_piece`. It's best not to set instance variables within a models methods. Can make the state of the object confusing, since it changes based on one of these methods being called (more prone to bugs :smile:)

You could do that by having `in_check?` either return the `threatening_piece`. Or return nil.

Then in your `checkmate?` method. You could do something like...

```
def checkmate?(color)
  threatening_piece = in_check?(color)
  return false unless threatening_piece

  ...
  
end
```